### ISSUE TITLE
@SneakyThrows can crash javac

### ISSUE THREAD

**Issue by lombokissues** [rspilker/trackertest#166]
*Migrated from Google Code (issue 56)*

**Comment by lombokissues** [comment]
:bust_in_silhouette: **lahoda** &nbsp; :clock8: Sep 29, 2009 at 10:38 UTC

Consider this code:
------------------------------------------------------
package a;

import java.io.IOException;
import lombok.SneakyThrows;

public class Main {

    public static void main(String[] args) {
        new Runnable() {
            @SneakyThrows(IOException.class)
            public void run() {
                throw new IOException();
            }
        };
    }
}
---------------------------------------------------

When building this code with javac, the attached exception occurs.

This is (AFAIK) caused by the fact that bodies of methods are inspected in
the attribute phase, but annotation processing is performed after the enter
phase. Synthetic &quot;value=&quot; assignment is generated during enter phase for
annotations on members/classes, but is generated during attribute phase for
the annotation on a method inside anonymous innerclass (see
MemberEnter.visitMethodDef/MemberEnter.enterAnnotations/Annotate.enterAnnotation).

**Comment by lombokissues** [comment]
:bust_in_silhouette: **lahoda** &nbsp; :clock8: Sep 29, 2009 at 10:38 UTC

[:link: lombok-javac-sneaky-throws-crash.txt](https://projectlombok.org/googlecode-issue-attachments/428476636308760803/lombok-javac-sneaky-throws-crash.txt) <a href=https://projectlombok.org/googlecode-issue-attachments/view/428476636308760803/lombok-javac-sneaky-throws-crash.txt" target="_blank"><img class="emoji" title="View file" alt="View file" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4c4.png" height="20" width="20" align="absmiddle"></a>

**Comment by lombokissues** [comment]
:bust_in_silhouette: **reinierz** &nbsp; :clock8: Oct 16, 2009 at 09:53 UTC

Heh, nice find, lahoda. Actually, it's a (somewhat rare) braindead design in the javac AST. For some reason on 
normal annotations, the type of the annotation parameters is a JCAssign, even if you're using the shorthand 
version (@Foo(bar) instead of @Foo(value=bar)), but when annotating method-local classes, you just get the 
righthand side node (so, JCNewArray or the appropriate literal type, pretty much). No idea why, but good job 
on finding it - with the stacktrace, it was easy to find, and fix.


Heh, after fixing it, I did a more thorough check of why that code was in there in the first place, and, turns 
out, there's no real reason for it. Removed the whole shebang - that logic has been moved into 
AnnotationValues, which is a lot smarter about things and wasn't suffering from this problem, showing how 
the forethought of separating this stuff out was good to begin with :P

Fixed in commit fdcdcc89d8ac081cb214aab6bb70f9d75101e865 and should be rolled out in v0.9
